volatility3/framework/plugins/windows/psxview.py

```diff
@@ -0,0 +1,279 @@
+
+    _required_framework_version = (2, 0, 0)
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.VersionRequirement(
+                name="info", component=info.Info, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="pslist", component=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="psscan", component=psscan.PsScan, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="thrdscan", component=thrdscan.ThrdScan, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="handles", component=handles.Handles, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="sessions", component=sessions.Sessions, version=(0, 0, 0)
+            ),
+            requirements.BooleanRequirement(
+                name="identify-expected",
+                description='In the plugin\'s output, replace false with \
+                                                normal where false is the expected result for a Windows machine running normally. \
+                                                Keep in mind that this plugin uses simple checks to identify "normal" behavior, \
+                                                so you may want to double-check the legitimacy of these processes yourself.',
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="physical-offsets",
+                description="List processes with phyiscall offsets instead of virtual offsets.",
+                optional=True,
+            ),
+        ]
+
+    def proc_name_to_string(self, proc):
+        return proc.ImageFileName.cast(
+            "string", max_length=proc.ImageFileName.vol.count, errors="replace"
+        )
+
+    def is_ascii(self, str):
+        return str.split(".")[0].isalnum()
+
+    def filter_garbage_procs(self, proc_list):
+        return [
+            p
+            for p in proc_list
+            if p.is_valid() and self.is_ascii(self.proc_name_to_string(p))
+        ]
+
+    def translate_offset(self, offset):
+        if self.config["physical-offsets"]:
+            return offset
+
+        kernel = self.context.modules[self.config["kernel"]]
+        layer_name = kernel.layer_name
+
+        try:
+            offset = list(
+                self.context.layers[layer_name].mapping(offset=offset, length=0)
+            )[0][2]
+        except:
+            # already have physical address
+            pass
```

Comment:
I'm not sure that this failing guarantees the offset was physical, I'd probably let the exception bubble up if you're not sure whether you're right or not.

Reply:
I agree. However, if I let the exception bubble up, it will page fault on the memory dumps I'm using to test. The address causing the page faults appears similar to the physical offsets reported by the original Psxview, so I assumed it was already physical. I'm not sure that assumption makes sense, but either way it seems necessary to not let the exception bubble up here. Do you have any other ideas on how to handle it?

Reply:
If you do want to catch it, at the very least there should be a debug log to say that something went wrong and then skip over it (and the exception should be more specific, probably an invalid address exception or something (to avoid catching other problems and ignoring them).

Reply:
I missed this in the commit I just added. Will look at it again